File: chazm-model/src/main/java/io/github/runtimemodels/chazm/event/ModeratesEventFactory.java
package io.github.runtimemodels.chazm.event;

import runtimemodels.chazm.api.relation.Moderates;

/**
 * The {@linkplain ModeratesEventFactory} interface defines the API for constructing {@linkplain ModeratesEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface ModeratesEventFactory {

    /**
     * Constructs an {@linkplain ModeratesEvent}.
     *
     * @param category  the {@linkplain EventCategory}.
     * @param moderates the {@linkplain Moderates}.
     * @return a {@linkplain ModeratesEvent}.
     */
    ModeratesEvent build(EventCategory category, Moderates moderates);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/ModeratesEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Moderates
import runtimemodels.chazm.model.event.EventType

/**
 * The [ModeratesEventFactory] interface defines the API for constructing [ModeratesEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface ModeratesEventFactory {
    /**
     * Constructs an [ModeratesEvent].
     *
     * @param category  the [EventType].
     * @param moderates the [Moderates].
     * @return a [ModeratesEvent].
     */
    fun build(category: EventType, moderates: Moderates): ModeratesEvent
}
